### PR TITLE
Add bash completions

### PR DESCRIPTION
**Pull request by MarkusGordathian**
Create the /completions directory and add a completion script for bash

This addresses #127, for bash at least

Requesting advice on how to fully implement this into playerctl - whether it should be hard-coded into the binary or read from a file.

### CODE REVIEW

File: completions/completions.bash [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		loop)
+			COMPREPLY=($(compgen -W "none track playlist" -- "$cur"))
+			return 0
+			;;
+		shuffle)
+			COMPREPLY=($(compgen -W "on off" -- "$cur"))
+			return 0
+			;;
+		-p|--player=|-i|--ignore-player=)
+			# TODO list supported players here?
```
Comment (acrisci) [inline completions/completions.bash]: It's a comma separated list of the output of `playerctl --list-all`

**Comment by acrisci** [comment]
I want to keep it in the repo as a file so it's easier to edit and I might want to distribute the file as well for the use case where people want to source the file instead of run the binary.

To do this, there are two options. You could either template the completions in with meson or [follow these instructions](https://www.linuxjournal.com/content/embedding-file-executable-aka-hello-world-version-5967) to convert the files into something that can be linked in.

**Comment by MarkusGordathian** [comment]
Applied requested changes to the completion script and edited build files to install to `/etc/bash_completion.d/playerctl.bash`

File: completions/meson.build [hunk 2]
```diff
@@ -0,0 +1 @@
+install_data(sources : 'playerctl.bash', install_dir : '/etc/bash_completion.d')
```
Comment (acrisci) [inline completions/meson.build]: This should be optional and probably doesn't go here. Package maintainers might have some guidelines for how to include these.

I like [this example](https://github.com/swaywm/sway/blob/f79f00dbfb678c25e4c9ed423ef808fded312c33/meson.build#L241).

**Comment by acrisci** [comment]
Hey @knowledgejunkie we're getting bash completions. Do you know of any guidelines on how to package them?

**Comment by knowledgejunkie** [comment]
I'm not aware of specific guidelines on how to package them in source. Many projects that include bash/zsh completions (in e.g. contrib/completions/*) leave it up to the end user to manually install them (per user or system-wide). Other projects might make it part of the `make install` process to make them available to all users.

Distros usually have special tooling in the packaging build system to ensure completions get installed into the correct location (e.g. on Debian we have the dh_bash-completion tool which installs to /usr/share/bash-completion/completions/) to make them available to all users on the system.

**Comment by acrisci** [comment]
Thanks. From what I can see based on that discussion, bash completions are usually supposed to go in a separate package. So we won't install them by default and leave it up to the packagers to decide how to handle it.

**Comment by MarkusGordathian** [comment]
Sounds good to me. Took the completions out of the package.

**Comment by acrisci** [comment]
:+1: :+1:

**Pull request merged**